exercises/03.async/01.solution.await/README.mdx

```diff
@@ -17,13 +17,13 @@ If I run the tests now, I can correctly see the assertion on `greetByResponse()`
 Error: Expected Hello, undefined! Happy, Monday. to equal to Hello, Patrick! Happy, Monday.
 ```
 
-The rejected assertion message seems to include `undefined` instead of the user's first name. Maybe the way we access the name on the user response is incorrect? I jump to the `greetByResponse()` implementation and, indeed, I can see we are accessing `user.name` instead of `user.firstName`.
+The rejected assertion message seems to include `undefined` instead of the user's first name. Maybe the way we access the name on the user response is incorrect? I jump to the `greetByResponse()` implementation and, indeed, I can see we are accessing `user.firstName` instead of `user.name`.
```

Comment:
This was correct. `user.name` is the mistake in the exercise, and `user.firstName` is the right thing to do in the code. I believe you can remove this change.

Reply:
I will double check to make sure diff isn't deceiving me...

Reply:
I confirm `user.firstName` is the intended implementation. So the only change is to adjust the mock `Response.json()` constructed in the test (it's a problem with the test setup, in particular how the mocked response is prepared).

Reply:
Hi, 
When looking at [exercises/03.async/01.problem.await/greet.ts](https://github.com/epicweb-dev/testing-fundamentals/blob/dc08329ed013d831f68d83b82f1108b1e89ae6f2/exercises/03.async/01.problem.await/greet.ts#L9) I already see `return greet(user.firstName)`. And in [exercises/03.async/01.problem.await/greet.test.ts](https://github.com/epicweb-dev/testing-fundamentals/blob/dc08329ed013d831f68d83b82f1108b1e89ae6f2/exercises/03.async/01.problem.await/greet.test.ts#L20) I see the data as `const response = Response.json({ name: 'Patrick' })`.

I doubled checked this and to get the tests to pass, I had to make this change in `greet.ts` 
```
export async function greetByResponse(response: Response) {
	const user = await response.json()
	return greet(user.**firstName**)
}
```
To
```
export async function greetByResponse(response: Response) {
	const user = await response.json()
	return greet(user.**name**)
}
```

Reply:
Ah. So what was unclear to me from the Solution was that it was showing the change was needed in the `greetByResponse()` function, as:
<img width="541" alt="Screenshot 2024-02-27 at 7 27 32 AM" src="https://github.com/epicweb-dev/testing-fundamentals/assets/4021179/7b3129ae-93ca-47b3-95c5-9ed7c67f70ce">

If the intention is to change it here, then I had to change `return greet(user.firstName)` to `return greet(user.name)` to get the tests to pass.

However if you do want `greetByResponse()` to have `return greet(user.firstName)` then we change the diff image to have the user change the `const response = Response.json({ name: 'Patrick' })` in the test to `const response = Response.json({ firstName: 'Patrick' })

Reply:
In the problem setup [exercises/03.async/01.problem.await/greet.ts](https://github.com/epicweb-dev/testing-fundamentals/blob/dc08329ed013d831f68d83b82f1108b1e89ae6f2/exercises/03.async/01.problem.await/greet.ts#L9) `greetByResponse()` already has `user.firstName`

Reply:
You make a great point! Let me check the problem part to be correct here.